marlowe-runtime/indexer/Language/Marlowe/Runtime/Indexer/Party.hs

```diff
@@ -110,20 +113,24 @@ loadContracts fromSlot =
           AND contractTxOutPartyRole.txId IS NULL
       |]
 
-decodeContractTxOut :: (ByteString, Int16, ByteString, Int16, ByteString, Int64, ByteString) -> ContractTxOut
-decodeContractTxOut (txId, txIx, createTxId, createTxIx, contract, slotNo, rolesCurrency) =
+decodeContractTxOut
+  :: (ByteString, Int16, ByteString, Int16, ByteString, ByteString, Int64, ByteString) -> ContractTxOut
+decodeContractTxOut (txId, txIx, createTxId, createTxIx, contract, state, slotNo, rolesCurrency) =
   ContractTxOut
     { contractOut = TxOutRef (TxId txId) (fromIntegral txIx)
     , contractId = ContractId $ TxOutRef (TxId createTxId) (fromIntegral createTxIx)
     , contract = fromJust $ fromDatum $ runGet get $ LBS.fromStrict contract
+    , state = fromJust $ fromDatum $ runGet get $ LBS.fromStrict state
     , slotNo = fromIntegral slotNo
     , rolesCurrency
     }
 
 toEntries :: ContractTxOut -> (Max SlotNo, [ContractTxOutParty])
 toEntries ContractTxOut{..} =
   ( Max slotNo
-  , [ContractTxOutParty{..} | party <- Set.toList $ extractAll contract]
+  , nub $
+      [ContractTxOutParty{..} | party <- Set.toList $ extractAll contract]
```

Comment:
Should we include choices as well? Whenever we have initial input applications this could be even more relevant.

Reply:
Well spotted; I forgot about those!